src/core/thread/router_table.hpp

```diff
@@ -110,7 +114,63 @@ class RouterTable : public InstanceLocator
          */
         Router *GetRouter(void) { return mRouter; }
```

Comment:
Is this method still necessary?

Reply:
I think it can be useful to keep it. Mainly since we have the `Iterator` as a public nested type of `RouterTable`, and it can be used directly in future (though at moment all modules use it indirectly through ranged-based  "for" loop).

In the case of direct use (i.e. another module defines `RouterTable::Iterator iter` and then use `iter`), having `GetRouter()` method to get a pointer to the `Router` entry can be helpful. We can use de-reference `operator*` to get a reference to the `Router` entry but `GetRouter()` (returning a pointer) would be more flexible and can be used even when the entry is `null`.

I see `GetRouter()` the same way as `IsDone()` which is not used in current code, but we want to keep it so that we have a proper `Iterator` definition.

Reply:
Since we can always call `IsDone()` to check if the iterator is at the end, there is no need to call `GetRouter()` because we can always call `&(*iter)` after we checked `!iter.IsDone()`. 
Same issue for the `Reset()` since it's never called. Further more I think it's better to create a new Iterator rather than reusing existing Iterator by calling `Reset()`. 

Is it a better to elide unnecessary code and only add when we really need it in the future?

Reply:
I agree not keeping unused code is a good idea. I removed them in new pushed commit